ionoscloud/resource_backup_unit.go

```diff
@@ -346,3 +347,33 @@ func setBackupUnitData(d *schema.ResourceData, backupUnit *ionoscloud.BackupUnit
 	}
 	return nil
 }
+
+// BackupUnitFindByID it's a custom function that returns a backup unit using its ID. The difference
+// between this function and the client BackupUnitsApi.BackupunitsFindById is that this function
+// retrieves the whole list of backup units and then iterates over them, returning the backup unit
+// when the ID is matched. This function was implemented because the response values in
+// the "BackupUnitsApi.BackupunitsFindById" method are invalid, but the values in
+// "BackupUnitsApi.BackupunitsGet" method are valid. This is a workaround for the API problem with
+// invalid values in the response for BackupUnitsApi.BackupunitsFindById.
```

Comment:
(nitpick) This is well detailed and explains all the context but honestly this could be simplified, like

```
BackupUnitFindByID simulates a FindByID function by filtering backup units from BackupunitsGet using the given ID.
This is done because of a temporary bug in the API with the regular FindByID function
```

Reply:
(comments should explain WHY it exists, the code in the function itself is readable and people can tell that the whole list of backups is returned and iterated over etc)

Reply:
@avirtopeanu-ionos I agree, I'll make the change, your suggestion is clearer.

Reply:
@avirtopeanu-ionos done in e4bb248a188c64025c74ce34b42a02d51ef8020f